plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenFacetImpl.java

```diff
@@ -160,7 +160,16 @@ public Content get(final MavenPath path) throws IOException {
     if (asset == null) {
       return null;
     }
+    return toContent(asset);
+  }
+
+  private Content toContent(final Asset asset) {
```

Comment:
This method's only used above, so you could just fold it into `get(...)` and avoid the extra call to get the current TX.